yarn-project/aztec.js/src/contract/contract_function_interaction.ts

```diff
@@ -70,42 +70,33 @@ export class ContractFunctionInteraction extends BaseContractInteraction {
    * Differs from prove in a few important ways:
    * 1. It returns the values of the function execution
    * 2. It supports `unconstrained`, `private` and `public` functions
-   * 3. For `private` execution it:
-   * 3.a SKIPS the entrypoint and starts directly at the function
-   * 4. For `public` execution it:
-   * 4.a Removes the `txRequest` value after ended simulation
-   * 4.b Ignores the `from` in the options
    *
    * @param options - An optional object containing additional configuration for the transaction.
    * @returns The result of the transaction as returned by the contract function.
    */
   public async simulate(options: SimulateMethodOptions = {}): Promise<any> {
     if (this.functionDao.functionType == FunctionType.UNCONSTRAINED) {
-      return this.wallet.viewTx(this.functionDao.name, this.args, this.contractAddress, options.from);
+      return this.wallet.simulateUnconstrained(this.functionDao.name, this.args, this.contractAddress, options?.from);
     }
 
-    if (this.functionDao.functionType == FunctionType.SECRET) {
-      const nodeInfo = await this.wallet.getNodeInfo();
-      const packedArgs = PackedValues.fromValues(encodeArguments(this.functionDao, this.args));
-      const gasSettings = options.gasSettings ?? GasSettings.simulation();
+    const txRequest = await this.create();
+    // const from =
+    //   this.functionDao.functionType == FunctionType.SECRET ? options.from ?? this.wallet.getAddress() : undefined;
 
-      const txRequest = TxExecutionRequest.from({
-        firstCallArgsHash: packedArgs.hash,
-        origin: this.contractAddress,
-        functionData: FunctionData.fromAbi(this.functionDao),
-        txContext: new TxContext(nodeInfo.chainId, nodeInfo.protocolVersion, gasSettings),
-        argsOfCalls: [packedArgs],
-        authWitnesses: [],
-      });
-      const simulatedTx = await this.wallet.simulateTx(txRequest, true, options.from ?? this.wallet.getAddress());
-      const flattened = simulatedTx.privateReturnValues;
-      return flattened ? decodeReturnValues(this.functionDao, flattened) : [];
-    } else {
-      const txRequest = await this.create();
-      const simulatedTx = await this.wallet.simulateTx(txRequest, true);
-      this.txRequest = undefined;
-      const flattened = simulatedTx.publicOutput?.publicReturnValues;
-      return flattened ? decodeReturnValues(this.functionDao, flattened) : [];
-    }
+    const simulatedTx = await this.wallet.simulateTx(txRequest, true, options?.from);
+
+    this.txRequest = undefined;
+
+    simulatedTx.tx;
```

Comment:
Huh 👀?

Reply:
Probably leftovers from your code 🤣 removing the undefined!